halo2_proofs/src/plonk/permutation.rs

```diff
@@ -75,6 +78,23 @@ pub(crate) struct VerifyingKey<C: CurveAffine> {
     commitments: Vec<C>,
 }
 
+impl<C: CurveAffine> VerifyingKey<C> {
+    pub(crate) fn write<W: io::Write>(&self, writer: &mut W) -> io::Result<()> {
+        for commitment in &self.commitments {
+            writer.write_all(commitment.to_bytes().as_ref())?;
+        }
+
+        Ok(())
+    }
+
+    pub(crate) fn read<R: io::Read>(reader: &mut R, argument: &Argument) -> io::Result<Self> {
```

Comment:
```suggestion
    pub(crate) fn read<R: io::Read>(reader: &mut R, argument: &Argument) -> io::Result<Self> {
    ```suggestion
        let mut num_commitments_le_bytes = [0u8; 4];
        reader.read_exact(&mut num_commitments_le_bytes)?;
        if argument.columns.len() != u32::from_le_bytes(num_commitments_le_bytes) {
            return io::Error::new(io::ErrorKind::InvalidData, "unexpected number of column commitments");
        }
```
```